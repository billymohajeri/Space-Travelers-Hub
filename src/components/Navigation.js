import { NavLink } from 'react-router-dom';
import '../styles/Navigation.css';
import logo from '../Assets/planet.png';

const Navigation = () => (
  <nav className="navmenu-desktop">
    <img src={logo} alt="logo" className="logo-img" />
    <h1 className="logo">Space Travelers&apos; Hub</h1>
    <ul className="nav-items">
      <li className="nav-item">
        <NavLink
          to="/"
          className={({ isActive }) => (isActive ? 'activeLink' : 'nonActiveLink')}
        />
      </li>
      <li className="nav-item">
        <NavLink
          to="rockets"
          className={({ isActive }) => (isActive ? 'activeLink' : 'nonActiveLink')}
        >
          Rockets
        </NavLink>
      </li>
      <li className="nav-item">
        <NavLink
          to="missions"
          end
          className={({ isActive }) => (isActive ? 'activeLink' : 'nonActiveLink')}
        >
          Missions
        </NavLink>
      </li>
      <li className="nav-divider">|</li>
      <li className="nav-item">
        <NavLink
          to="my-profile"
          className={({ isActive }) => (isActive ? 'activeLink' : 'nonActiveLink')}
        >
          My Profile
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navigation;
