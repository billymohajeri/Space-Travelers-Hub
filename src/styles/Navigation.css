.logo {
  font-family: 'Montserrat', sans-serif;
  font-size: 2rem;
  margin-left: 2%;
}

.logo-img {
  width: 100px;
}

.navmenu-desktop {
  padding: 24px 4%;
  background-color: #fff;
  border-bottom: 1px solid #e8e8e8;
  display: flex;
  align-items: center;
}

.nav-items {
  display: flex;
  align-items: center;
  list-style: none;
  margin-top: 0;
  margin-bottom: 0;
  gap: 40px;
  margin-left: auto;
}

.nav-item {
  font-family: 'Montserrat', sans-serif;
  font-size: 0.813rem;
  color: #121212;
  letter-spacing: 1.9px;
}
