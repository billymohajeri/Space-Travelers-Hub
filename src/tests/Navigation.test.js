import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navigation from '../components/Navigation';

it('renders correctly', () => {
  const navigation = renderer
    .create(<Router><Navigation /></Router>)
    .toJSON();
  expect(navigation).toMatchSnapshot();
});
