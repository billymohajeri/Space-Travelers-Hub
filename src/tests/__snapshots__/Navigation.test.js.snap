// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navmenu-desktop"
>
  <img
    alt="logo"
    className="logo-img"
    src="planet.png"
  />
  <h1
    className="logo"
  >
    Space Travelers' Hub
  </h1>
  <ul
    className="nav-items"
  >
    <li
      className="nav-item"
    >
      <a
        aria-current="page"
        className="activeLink"
        href="/"
        onClick={[Function]}
      />
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nonActiveLink"
        href="/rockets"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nonActiveLink"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li
      className="nav-divider"
    >
      |
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nonActiveLink"
        href="/my-profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
